lib/open_food_network/order_cycle_permissions.rb

```diff
@@ -154,49 +151,30 @@ def editable_variants_for_incoming_exchanges_from(producer)
       end
     end
 
+    def all_incoming_editable_variants
+      valid_suppliers = visible_enterprises.map do |enterprise|
+        enterprise.id if user_manages_coordinator_or(enterprise)
+      end
+
+      Spree::Variant.includes(product: :supplier).
+        joins(:product).where('spree_products.supplier_id IN (?)', valid_suppliers)
```

Comment:
you should be able to rely on AR syntax to get the supplier id but also isn't the following query enough?

```rb
Spree::Variant
  .joins(:product)
  .where(supplier_id: valid_suppliers)
```

given `supplier_id` is a column of `spree_products`.